sklearn/feature_selection/base.py

```diff
@@ -120,3 +120,67 @@ def inverse_transform(self, X):
         Xt = np.zeros((X.shape[0], support.size), dtype=X.dtype)
         Xt[:, support] = X
         return Xt
+
+
+def feature_wise_scorer(score_func):
+    """ A wrapper function around score functions.
+
+    Parameters
+    ----------
+    score_func : callable
+        Function taking array(s) X and/or y, and returning a pair of arrays
+        (scores, pvalues) or a single array with scores.
+
+    Returns
+    -------
+    scores : array, shape = [n_features,]
+        Score values returned by the scoring function.
+    p_vals : array, shape = [n_features,]
+        The set of p-values returned by the scoring function.
+
+    Notes
+    -----
+    This wrapper function wraps around scoring functions like `spearmanr`,
+    `pearsonr` etc. from the scipy.stats module and makes it usable for
+    feature selection algorithms like `SelectKBest`.
+
+    Example
+    -------
+    >>> from sklearn.feature_selection import feature_wise_scorer, SelectKBest
+    >>> from scipy.stats import kendalltau, pearsonr, spearmanr
+    >>> from sklearn.datasets import make_classification
+    >>> X, y = make_classification(random_state=0)
+    >>> skb1 = SelectKBest(feature_wise_scorer(spearmanr), k=10)
+    >>> skb2 = SelectKBest(feature_wise_scorer(pearsonr), k=10)
+    >>> skb3 = SelectKBest(feature_wise_scorer(kendalltau), k=10)
+    >>> skb1.fit(X, y) #doctest: +SKIP
+    >>> skb2.fit(X, y) #doctest: +SKIP
+    >>> skb3.fit(X, y) #doctest: +SKIP
+    >>> new_X1 = skb1.transform(X) #doctest: +SKIP
+    >>> new_X2 = skb2.transform(X) #doctest: +SKIP
+    >>> new_X3 = skb3.transform(X) #doctest: +SKIP
+    """
+    def call_scorer(*args, **kwargs):
+        X = args[0]
+        y = args[1]
+        scores = []
+        p_vals = []
+
+        for i in range(0, X.shape[1]):
+            score_func_ret = score_func(X[:, i], y, **kwargs)
+            if isinstance(score_func_ret, (list, tuple)):
+                score, p_val = score_func_ret
+                p_vals.append(p_val)
+            else:
+                score = score_func_ret
+                p_vals = None
+            scores.append(abs(score))
```

Comment:
I see why you're using the `abs` function here, since this would weigh a correlation score of -1 the same as +1. 

However, it would be nice if you documented that you'd weigh two features equally if their absolute values were the same. I also don't know whether you can say that _for any metric_, the absolute value of said metric corresponds to an appropriate score.